target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts

```diff
@@ -39,12 +39,150 @@ import {
+   *
+   * @example
+   * ```typescript
+   * const priceUpdateData = await priceServiceConnection.getLatestVaas([
+   *    SOL_PRICE_FEED_ID,
+   *    ETH_PRICE_FEED_ID,
+   * ]);
+   *
+   * const transactionBuilder = pythSolanaReceiver.newTransactionBuilder();
+   * await transactionBuilder.withPostPartiallyVerifiedPriceUpdates(priceUpdateData);
+   * await transactionBuilder.withPriceConsumerInstructions(...)
+   * ...
+   * ```
+   */
+  async withPostPartiallyVerifiedPriceUpdates(priceUpdateDataArray: string[]) {
+    const {
+      postInstructions,
+      priceFeedIdToPriceUpdateAccount,
+      closeInstructions,
+    } = await this.pythSolanaReceiver.buildPostPriceUpdateAtomicInstructions(
+      priceUpdateDataArray
+    );
+    this.closeInstructions.push(...closeInstructions);
+    this.priceFeedIdToPriceUpdateAccount = {
+      ...this.priceFeedIdToPriceUpdateAccount,
+      ...priceFeedIdToPriceUpdateAccount,
+    };
+    this.addInstructions(postInstructions);
+  }
+
+  /**
+   * Add instructions that consume price updates to the builder.
+   *
+   * @param getInstructions a function that given a map of price feed IDs to price update accounts, generates a series of instructions. Price updates get posted to ephemeral accounts and this function allows the user to indicate which accounts in their instruction need to be "replaced" with each price update account.
+   *
+   * @example
+   * ```typescript
+   * ...
+   * await transactionBuilder.withPostPriceUpdates(priceUpdateData);
+   * await transactionBuilder.withPriceConsumerInstructions(
+   *   async (
+   *     priceFeedIdToPriceAccount: Record<string, PublicKey>
+   *   ): Promise<InstructionWithEphemeralSigners[]> => {
+   *     return [
+   *       {
+   *         instruction: await myFirstPythApp.methods
+   *           .consume()
+   *           .accounts({
+   *              solPriceUpdate: priceFeedIdToPriceUpdateAccount[SOL_PRICE_FEED_ID],
+   *              ethPriceUpdate: priceFeedIdToPriceUpdateAccount[ETH_PRICE_FEED_ID],
+   *           })
+   *           .instruction(),
+   *         signers: [],
+   *       },
+   *     ];
+   *   }
+   * );
+   * transactionBuilder.withCloseInstructions();
+   * ```
+   */
+  async withPriceConsumerInstructions(
+    getInstructions: (
+      priceFeedIdToPriceUpdateAccount: Record<string, PublicKey>
+    ) => Promise<InstructionWithEphemeralSigners[]>
+  ) {
+    this.addInstructions(
+      await getInstructions(this.priceFeedIdToPriceUpdateAccount)
+    );
+  }
+
+  /**
+   * Add instructions to close the encoded vaa accounts and the price update accounts created by `withPostPriceUpdates` and `withPostPartiallyVerifiedPriceUpdates` to reclaim rent.
+   */
+  withCloseInstructions() {
+    this.addInstructions(this.closeInstructions);
+  }
```

Comment:
The pattern here for closing the price accounts is confusing. I think a simpler approach would be to:

1. Pass a boolean `closeAccount` flag to `withPostPriceUpdates` (etc) above
2. Automatically append `this.closeInstructions` to the instruction list in `getVersionedTransactions` (or anywhere the builder is built)
3. Delete this method

Reply:
Done